msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java

```diff
@@ -1547,6 +1562,57 @@ public Object doInHibernate(Session session) throws HibernateException, SQLExcep
 
 			return (Topic) getHibernateTemplate().execute(hcb);
 		}
+
+		public List<Topic> getTopicsInSite(final String contextId)
+		{
+			return getTopicsInSite(contextId, false);
+		}
+
+		public List<Topic> getTopicsInSite(final String contextId, boolean anonymousOnly)
+		{
+			if (contextId == null)
+			{
+				throw new IllegalArgumentException("Null Argument");
+			}
+
+			final String query = anonymousOnly ? "findAnonymousTopicsInSite" : "findTopicsInSite";
+
+			HibernateCallback<List<Topic>> hcb = new HibernateCallback<List<Topic>>()
+			{
+				public List<Topic> doInHibernate(Session session) throws HibernateException, SQLException
+				{
+					Query q = session.getNamedQuery(query);
+					q.setParameter("contextId", contextId, Hibernate.STRING);
+					return q.list();
+				}
+			};
+
+			List<Topic> topicList = new ArrayList<Topic>();
+			List resultSet = (List) getHibernateTemplate().execute(hcb);
+			for (Object objResultArray : resultSet)
```

Comment:
IIRC I was getting a List of Object arrays, and each array contained [Topic, Forum, Area]. I think Hibernate sees the joins and includes it in the result set.